### PR TITLE
feat: add crosschain erc20 interface

### PR DESCRIPTION
**Pull request by agusduha**

### CODE REVIEW

File: packages/contracts-bedrock/src/L2/SuperchainERC20.sol [hunk 1]
```diff
@@ -10,7 +11,7 @@ import { ERC20 } from "@solady/tokens/ERC20.sol";
 /// @notice SuperchainERC20 is a standard extension of the base ERC20 token contract that unifies ERC20 token
 ///         bridging to make it fungible across the Superchain. This construction allows the SuperchainERC20Bridge to
 ///         burn and mint tokens.
-abstract contract SuperchainERC20 is ERC20, ISuperchainERC20Extension, ISemver {
+abstract contract SuperchainERC20 is ERC20, ICrosschainERC20, ISuperchainERC20Errors, ISemver {
```
Comment (0xng) [inline packages/contracts-bedrock/src/L2/SuperchainERC20.sol]: isn't this the same as inheriting from `ISuperchainERC20Exension` directly?

File: packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/// @title ICrosschainERC20
+/// @notice This interface is a standard for crosschain ERC20 transfers.
```
Comment (0xng) [inline packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol]:
```suggestion
/// @notice Defines the interface for crosschain ERC20 transfers.
```

File: packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+interface ICrosschainERC20 {
+    /// @notice Emitted whenever tokens are minted by a crosschain transfer.
```
Comment (0xng) [inline packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol]: If you prefer active voice: 
```suggestion
    /// @notice Emitted when a crosschain transfer mints tokens.
```

same for the burn event

File: packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    /// @param account Address of the account tokens are being minted for.
+    /// @param amount  Amount of tokens minted.
+    event CrosschainMinted(address indexed account, uint256 amount);
```
Comment (0xng) [inline packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol]: is account pattern matching a previous pattern from them? does it make sense to call it `to` like the function argument name? or `recipient`?

File: packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    /// @notice Emitted whenever tokens are burned by a crosschain transfer.
+    /// @param account Address of the account tokens are being burned from.
+    /// @param amount  Amount of tokens burned.
+    event CrosschainBurnt(address indexed account, uint256 amount);
+
+    /// @notice Allows to mint tokens through a crosschain transfer.
```
Comment (0xng) [inline packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol]: Alternatives: 
1.  Mints tokens due after a crosschain transfer.
2.  Mints tokens via crosschain transfer.

File: packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    /// @param _to     Address to mint tokens to.
+    /// @param _amount Amount of tokens to mint.
+    function __crosschainMint(address _to, uint256 _amount) external;
+
+    /// @notice Allows to burn tokens through a crosschain transfer.
```
Comment (0xng) [inline packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol]: Alternatives: 
1.  Burns tokens to initiate a crosschain transfer.
2.  Burns tokens via crosschain transfer.

File: packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol (hunk 4 above)
Comment (agusduha) [reply packages/contracts-bedrock/src/L2/interfaces/ICrosschainERC20.sol]: they have them like this in `OptimismMintable` but i would prefer to use `to` and `from` in both events

**Review by 0xng — APPROVED** [review]

**Pull request merged**